### PR TITLE
 Dynamic fee strategies with Fee Router

### PR DESCRIPTION
**Pull request by enemycnt**
## Related Issue Or Context

Closes: #3

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: packages/sdk/src/Sygma.ts [hunk 1]
```diff
@@ -273,27 +252,33 @@ import Connector from './connectors/Connectors';
...
+  public createProposalExecutionEventListener(chain: Directions, homeDepositNonce: number) {
+    const destination = this.bridgeSetup[chain]
     const bridge = this.bridges![chain]!;
     const proposalFilter = bridge.filters.ProposalExecution(null, null, null);
 
     const proposalExecutionEventListener = (callbackFn: any) =>
-      bridge.once(proposalFilter, (originDomainId, despositNonce, dataHash, tx) => {
-        callbackFn(originDomainId, despositNonce, dataHash, tx);
+      bridge.on(proposalFilter, (originDomainId, depositNonce, dataHash, tx) => {
+        console.log("🚀 ~ file: Sygma.ts ~ line 306 ~ Sygma ~ createProposalExecutionEventListener ~ homeDepositNonce", homeDepositNonce)
```
Comment (wainola) [inline packages/sdk/src/Sygma.ts:277]: remove logs

File: packages/sdk/src/Sygma.ts [hunk 2]
```diff
@@ -273,27 +252,33 @@ import Connector from './connectors/Connectors';
...
+        console.log("🚀 ~ file: Sygma.ts ~ line 312 ~ Sygma ~ bridge.on ~ depositNonce", depositNonce.toNumber())
```
Comment (wainola) [inline packages/sdk/src/Sygma.ts:278]: remove logs

File: packages/sdk/src/Sygma.ts [hunk 3]
```diff
@@ -302,22 +287,22 @@ import Connector from './connectors/Connectors';
...
   public removeProposalExecutionEventListener(chain: Directions) {
     const bridge = this.bridges![chain]!;
     const proposalFilter = bridge.filters.ProposalExecution(null, null, null);
-    return bridge.removeAllListeners(proposalFilter)
+    return bridge.removeAllListeners(proposalFilter);
   }
 
-  public destinationProposalExecutionEventListener(callback: any) {
-    return this.createProposalExecutionEventListener('chain2')(callback);
+  public destinationProposalExecutionEventListener(homeDepositNonce: number, callback: any) {
```
Comment (wainola) [inline packages/sdk/src/Sygma.ts:300]: is it possible here to type the `callback`?

File: packages/sdk/src/Sygma.ts [hunk 4]
```diff
@@ -435,28 +420,30 @@ import Connector from './connectors/Connectors';
...
+    const sender = await this.signers!.chain1?.getAddress() ?? ethers.constants.AddressZero;
 
     const feeData = calculateFeeData({
       provider,
       sender,
       recipientAddress,
       fromDomainID: parseInt(this.bridgeSetup.chain1.domainId),
       toDomainID: parseInt(this.bridgeSetup.chain2.domainId),
-      tokenResource: erc20Address,
+
```
Comment (wainola) [inline packages/sdk/src/Sygma.ts:445]: remove enter

File: packages/sdk/src/fee/feeOracle.ts [hunk 5]
```diff
@@ -43,11 +43,11 @@ export const createOracleFeeData = (
     toHex(oracleResponse.expirationTimestamp, 32).substr(2) + // timestamp: uint256
     toHex(oracleResponse.fromDomainID, 32).substr(2) + // fromDomainID: uint256
     toHex(oracleResponse.toDomainID, 32).substr(2) + // toDomainID: uint256
-    // addPadding(oracleResponse.resourceID, 32).substr(2); // resourceID: bytes32
-    addPadding(
-      tokenResource.substr(2) + toHex(oracleResponse.fromDomainID, 1).substr(2),
-      32,
-    ).substr(2);
+    (oracleResponse.resourceID).substr(2); // resourceID: bytes32
+    // addPadding(
```
Comment (wainola) [inline packages/sdk/src/fee/feeOracle.ts:47]: are we going to use this?

File: packages/sdk/src/fee/feeOracle.ts [hunk 6]
```diff
@@ -141,7 +142,12 @@ export const requestFeeFromFeeOracle = async ({
 }) => {
   try {
     const response = await fetch(
-      `${feeOracleBaseUrl}/v1/rate/from/${fromDomainID}/to/${toDomainID}/token/${resourceID}`,
+      `${feeOracleBaseUrl}/v1/rate/from/${fromDomainID}/to/${toDomainID}/resourceid/${resourceID}`,
```
Comment (wainola) [inline packages/sdk/src/fee/feeOracle.ts:145]: it seems here there is an issue with eslint. There is a new line before the string

**Review by wainola — CHANGES_REQUESTED** [review]

File: packages/sdk/src/fee/feeOracle.ts (hunk 6 above)
Comment (enemycnt) [reply packages/sdk/src/fee/feeOracle.ts:145]: It looks like github's highlight is a bit misleading here. 
`/token/` changed to `/resourceid/` at the end of the line

**Review by wainola — APPROVED** [review]

**Pull request merged**